wp/lib/bap_wp/src/runner.ml

```diff
@@ -0,0 +1,516 @@
+(***************************************************************************)
+(*                                                                         *)
+(*  Copyright (C) 2018/2019 The Charles Stark Draper Laboratory, Inc.      *)
+(*                                                                         *)
+(*  This file is provided under the license found in the LICENSE file in   *)
+(*  the top-level directory of this project.                               *)
+(*                                                                         *)
+(*  This work is funded in part by ONR/NAWC Contract N6833518C0107.  Its   *)
+(*  content does not necessarily reflect the position or policy of the US  *)
+(*  Government and no official endorsement should be inferred.             *)
+(*                                                                         *)
+(***************************************************************************)
+
+open !Core_kernel
+open Bap.Std
+open Bap_core_theory
+
+(* FIXME: can we remove this include? *)
+include Self()
+
+module Comp = Compare
+module Pre = Precondition
+module Env = Environment
+module Constr = Constraint
+module Params = Run_parameters
+
+(** The available options to be set. Each flag corresponds to a parameter in
+    the set with the BAP custom command line. *)
+type params = Params.t
+
+let spec_of_name (name : string) : Sub.t -> Theory.target -> Env.fun_spec option =
```

Comment:
I dunno. Thematic? I'm open to opinions. Generally, if it might be useful to someone else (and doesn't have circular dependencies), I'm open to moving it to Utils.

Reply:
I don't have any strong opinions. I was just curious. I'm all good with keeping this as is.

Reply:
I'll keep it as-is for now then. People should just be using the `Runner` module anyway.